File: Codeforces-567A.cpp
#include <iostream>
#include <algorithm>
#include <vector>
using namespace std;
 
 
int main()
{
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
 
	int n, maxd = 0, mind = 0;
	cin >> n;
	vector<int>city(n);
	
 
	for (int i = 0; i < n; i++) cin >> city[i];
 
 
	for (int i = 0; i < n; i++) {
		if (i == 0) {
			maxd =abs( city[n - 1] - city[i]);
			mind = abs(city[i + 1] - city[i]);
		}
		else if (i==(n-1)) {
			maxd = abs(city[n-1]-city[0]);
			mind = abs(city[n-1]-city[n-2]);
		}
		else {
			mind = min(abs(city[i+1]-city[i]),abs(city[i-1]-city[i]));
			maxd = max(abs(city[i]-city[0]), abs(city[n-1]-city[i]));
		}
		cout << mind << " " << maxd<<"\n";
	}
 
    
 
	return 0;
}

File: Codeforces-567A.java
import java.util.ArrayList;
import java.util.Comparator;
import java.util.Scanner;

public class Solution {
    static class City {
        int x;
        int min;
        int max;

        public City(int x) {
            this.x = x;
        }

        @Override
        public String toString() {
            return this.min + " " + this.max;
        }
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        Scanner s = new Scanner(System.in);
        int n = s.nextInt();
        ArrayList<City> cities = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            cities.add(new City(s.nextInt()));
        }
        s.close();

        City c0 = cities.get(0);
        c0.min = Math.abs(cities.get(1).x - c0.x);
        City cn = cities.get(n - 1);
        cn.min = Math.abs(cn.x - cities.get(n - 2).x);

        for(int i=0; i<n; i++){
            City current = cities.get(i);
            current.max = Math.max( Math.abs(current.x - c0.x),Math.abs(current.x - cn.x) );
            if(i!=0 && i!=n-1){
                current.min = Math.min( Math.abs(current.x - cities.get(i+1).x),Math.abs(current.x -cities.get(i-1).x));
            }
        }
        for(var city: cities){
            System.out.println(city.toString());
        }
    }
}